packages/app-content-pages/src/screens/Publications/PublicationsContainer.spec.js

```diff
@@ -1,49 +1,56 @@
+import { RouterContext } from 'next/dist/shared/lib/router-context'
+import Router from 'next/router'
+import { composeStory } from '@storybook/react'
+
+import Meta, { Default } from './Publications.stories.js'
+import projectAnnotations from '../../../.storybook/preview.js'
+
+function RouterMock({ children }) {
+  const mockRouter = {
+    locale: 'en',
+    push: () => {},
+    prefetch: () => new Promise((resolve, reject) => {}),
+    query: {
+      owner: 'test-owner',
+      project: 'test-project'
+    }
+  }
+
+  Router.router = mockRouter
+
+  return (
+    <RouterContext.Provider value={mockRouter}>
+      {children}
+    </RouterContext.Provider>
+  )
+}
 
 describe('Component > PublicationsContainer', function () {
-  before(function () {
-    wrapper = shallow(<PublicationsContainer publicationsData={DATA} />)
-    componentWrapper = wrapper.find(PublicationsComponent)
-  })
-
-  it('should render without crashing', function () {
-    expect(wrapper).to.be.ok()
-  })
-
-  describe('with no publications data', function () {
-    it('should render without crashing', function () {
-      const noDataWrapper = shallow(<PublicationsContainer publicationsData={undefined} />)
-      expect(noDataWrapper).to.be.ok()
-    })
-  })
-
-  it('should render the `Publications` component', function () {
-    expect(componentWrapper).to.have.lengthOf(1)
-  })
-
-  describe('with no active filters', function () {
-    it('should pass down the expected props', function () {
-      expect(componentWrapper.prop('data')).to.deep.equal(DATA)
-    })
+  const DefaultStory = composeStory(Default, Meta, projectAnnotations)
+  it('should have a sidebar with available filters', function () {
+    render(
+      <RouterMock>
+        <DefaultStory />
+      </RouterMock>
+    )
+    const categoryFilters = DefaultStory.args.publicationsData.map(
+      category => category.title
+    )
+    const sideBar = document.querySelector('aside')
+    const listedFilters = within(sideBar).getAllByRole('link')
+    expect(listedFilters.length).to.equal(categoryFilters.length + 1) // +1 to account for Show All
+    expect(listedFilters[1].textContent).to.equal(categoryFilters[0])
   })
 
-  describe('with an active filter', function () {
-    it('should pass down the expected props', function () {
-      const filters = componentWrapper.prop('filters')
-      const targetFilter = filters[2]
-      targetFilter.setActive()
-      componentWrapper = wrapper.find(PublicationsComponent)
-      const expectedResult = DATA.filter(t => t.title === targetFilter.name)
-      expect(componentWrapper.prop('data')).to.deep.equal(expectedResult)
-      expect(componentWrapper.prop('filters')[2].active).to.be.true()
-    })
+  it('should render all publications in data', function () {
+    render(
+      <RouterMock>
+        <DefaultStory />
+      </RouterMock>
+    )
+    const publications = screen.getAllByTestId('publication-test-element')
```

Comment:
IDs are unique in HTML, with each identifier only allowed to occur once in a document. It seems a bit counterintuitive to use test ID to group a set of elements here. That’s usually done in HTML with a CSS class.

Reply:
Not a blocking comment, by the way. This just struck me as odd.

Reply:
Definitely agreed that `id` is only allowed to occur once, and is that always true for `data-testid`? I was thinking of it like [data-](url).

I supposed I could use `className` here instead, but just wanted a confident way to count the number of publications rendered.

Reply:
I don’t think it’s a big deal. Just a bit strange.